lib/Runtime/Library/JavascriptLibraryBase.h

```diff
@@ -147,6 +150,8 @@ namespace Js
         PropertyId GetPropertyIdSymbolIterator() { return PropertyIds::_symbolIterator; };
         PropertyId GetPropertyIdSymbolToStringTag() { return PropertyIds::_symbolToStringTag; };
 
+        bool IsChakraEngine() const { return chakraEngine != nullptr; }
```

Comment:
Any reason for this being on the library rather than the script context?

Reply:
The GlobalObject <-> Library relationship (both of which are recycler-allocated, as opposed to the heap-allocated ScriptContext) is guaranteed to be alive at points in time where we want to verify whether the lifetime management is being done by the ChakraEngine path (which now is , and ScriptContext might not be accessible. To some extent, this might just be an arbitrary design decision, but I'd have to do a significant amount of verification work if I want to switch that around.